services/core/shared/core/compat/sdk_v5/blocks.js

```diff
@@ -115,49 +115,66 @@ const buildIndex = async (from, to) => {
+		logger.info(`Attempting to cache blocks ${offset + 1}-${offset + MAX_BLOCKS_LIMIT_PP}`);
 		// TODO: Revert to standard notation, once getBlocks is fully implemented
 		// const blocks = await getBlocks({
 		// 	limit: MAX_BLOCKS_LIMIT_PP,
 		// 	offset: offset - 1,
 		// 	sort: 'height:asc',
 		// });
 		const blocks = await getBlocks({
-			heightRange: { from: offset, to: offset + MAX_BLOCKS_LIMIT_PP },
+			heightRange: { from: offset + 1, to: offset + MAX_BLOCKS_LIMIT_PP },
 		});
+		await indexBlocks(blocks.data);
+
+		blocks.data = blocks.data.sort((a, b) => a.height - b.height);
+
 		const topHeightFromBatch = (blocks.data.pop()).height;
-		await blocksCache.set('lastIndexedHeight', topHeightFromBatch);
+		const bottomHeightFromBatch = (blocks.data.shift()).height;
+		const lowestIndexedHeight = await blocksCache.get('lowestIndexedHeight');
+		if (bottomHeightFromBatch < lowestIndexedHeight) await blocksCache.set('lowestIndexedHeight', bottomHeightFromBatch);
+		if (topHeightFromBatch > highestIndexedHeight) await blocksCache.set('highestIndexedHeight', topHeightFromBatch);
 	});
 	logger.info(`Finished building block index (${from}-${to})`);
 };
 
 const init = async () => {
 	try {
+		const genesisHeight = 1;
 		const currentHeight = (await coreApi.getNetworkStatus()).data.height;
 
-		let blockIndexLowerRange = config.indexNumOfBlocks > 0
-			? currentHeight - config.indexNumOfBlocks : 1;
-		const lastNumOfBlocks = await blocksCache.get('lastNumOfBlocks');
+		const blockIndexLowerRange = config.indexNumOfBlocks > 0
+			? currentHeight - config.indexNumOfBlocks : genesisHeight;
+		const blockIndexHigherRange = currentHeight;
+
+		const highestIndexedHeight = await blocksCache.get('highestIndexedHeight') || blockIndexLowerRange;
 
-		if (Number(lastNumOfBlocks) === Number(config.indexNumOfBlocks)) {
-			// Everything seems allright, continue at height where stopped last time
-			blockIndexLowerRange = await blocksCache.get('lastIndexedHeight');
-		} else {
+		const lastNumOfBlocks = await blocksCache.get('lastNumOfBlocks');
+		if (lastNumOfBlocks !== config.indexNumOfBlocks) {
 			logger.info('Configuration has been updated, re-index eveything');
 			await blocksCache.set('lastNumOfBlocks', config.indexNumOfBlocks);
-			await blocksCache.set('lastIndexedHeight', blockIndexLowerRange);
+			await blocksCache.set('lowestIndexedHeight', genesisHeight);
+			await blocksCache.set('highestIndexedHeight', currentHeight);
 		}
 
-		await buildIndex(blockIndexLowerRange, currentHeight);
+		await buildIndex(highestIndexedHeight, blockIndexHigherRange);
+
+		const lowestIndexedHeight = await blocksCache.get('lowestIndexedHeight') || genesisHeight;
+		if (blockIndexLowerRange < lowestIndexedHeight) {
+			// If the indexing is partially built
+			await buildIndex(blockIndexLowerRange, lowestIndexedHeight);
+		}
 	} catch (err) {
 		logger.warn('Unable to update block index');
 		logger.warn(err.message);
 	}
 };
 
-init();
+setTimeout(init, 5000);
```

Comment:
We need to make sure this is replaced with the `init()` function, similarly to SDKv4.